tutorials-v5/time-evolution/023_v5_paper-mesolve.md

```diff
@@ -0,0 +1,575 @@
+me_global_res = mesolve(H_weak, psi0, tlist, co_list, e_ops=[sz1, sz2])
+br_res = brmesolve(
+    H_weak,
+    psi0,
+    tlist,
+    e_ops=[sz1, sz2],
+    a_ops=[[sx1, power_spectrum], [sx2, power_spectrum]],
+)
+```
+
+```python
+plt.plot(tlist, me_local_res.expect[0], label=r"Local Lindblad")
+plt.plot(tlist, me_global_res.expect[0], "--", label=r"Dressed Lindblad")
+plt.plot(tlist, br_res.expect[0], ":", label=r"Bloch-Redfield")
+plt.title("Weak Coupling")
+plt.xlabel("Time")
+plt.ylabel(r"$\langle \sigma_z^{(1)} \rangle$")
+plt.legend()
+plt.show()
+```
+
+```python
+# strong coupling
+g = 2 * epsilon1
+H_strong = (epsilon1 / 2) * sz1 + (epsilon2 / 2) * sz2 + g * sx1 * sx2
+```
+
+```python
+# generate new collapse operators for weak coupling Hamiltonian
+all_energy, all_state = H_strong.eigenstates()
+co_list = make_co_list(all_energy, all_state)
+
+# time list with smaller steps
+tlist_fine = np.linspace(0, 40, 1000)
+```
+
+```python
+me_local_res = mesolve(H_strong, psi0, tlist_fine, c_ops, e_ops=[sz1, sz2])
+me_global_res = mesolve(H_strong, psi0, tlist_fine, co_list, e_ops=[sz1, sz2])
+br_res = brmesolve(
+    H_strong,
+    psi0,
+    tlist_fine,
+    e_ops=[sz1, sz2],
+    a_ops=[[sx1, power_spectrum], [sx2, power_spectrum]],
+)
+```
+
+```python
+plt.plot(tlist_fine, me_local_res.expect[0], label=r"Local Lindblad")
+plt.plot(tlist_fine, me_global_res.expect[0], "--", label=r"Dressed Lindblad")
+plt.plot(tlist_fine, br_res.expect[0], ":", label=r"Bloch-Redfield")
+plt.title("Strong Coupling")
+plt.xlabel("Time")
+plt.ylabel(r"$\langle \sigma_z^{(1)} \rangle$")
+plt.legend()
+plt.show()
+```
+
+### Manual Liouvillian Superoperator
+
+As mentioned before, QuTiP's master equation solver can be used to solve any other master equation.
+By using `spre()`, `spost()` and `sprepost()`, we can manually construct such equations.
+These functions specifically convert the operators from the original Hilbert space to operators in the double space which is internally used by QuTiP to optimize computations (more details on this can be found in the [QuTiPv5 paper](#References)).
+
+For example, the Lindbladian corresponding to the master equation of the previous example can be constructed manually via:
+
+
+```python
+lindbladian = -1.0j * (spre(H) - spost(H))
+for c in c_ops:
+    lindbladian += sprepost(c, c.dag())
+    lindbladian -= 0.5 * (spre(c.dag() * c) + spost(c.dag() * c))
+```
+
+```python
+manual_res = mesolve(lindbladian, psi0, tlist_fine, [], e_ops=[sz1, sz2])
+```
+
+```python
+plt.plot(tlist_fine, me_local_res.expect[0], label="i=1")
+plt.plot(tlist_fine, me_local_res.expect[1], label="i=2")
+plt.xlabel("Time")
+plt.ylabel(r"$\langle \sigma_z^{(i)} \rangle$")
+plt.legend()
+plt.show()
+```
+
+<!-- #region -->
+## Part 2: Time-Dependent Systems
+
+
+Finally, we compare the results with another `mesolve` considering the rotating-wave approximation, the Bloch-Redfield solver and the HEOMSolver.
```

Comment:
"Finally, we compare the results from another example, that of a driven system using, first, `mesolve` with and without the rotating-wave approximation (RWA) for the drive, and then without RWA with the Bloch-Redfield and HEOM solvers." (just for clarity)